Uplift/Models/Gym.swift

```diff
@@ -253,6 +253,29 @@ struct FacilityDetail {
             subfacilities = detailData.subFacilityNames.compactMap({ $0 })
         }
     }
+
+    func getEquipmentCategories() -> [EquipmentCategory] {
+       var equipmentDict = [String: [Equipment]]()
+        equipment.forEach { equipmentItem in
+            if equipmentDict[equipmentItem.equipmentType] != nil {
```

Comment:
This can be accomplished in one line with 

`equipementDict[equipmentItem.equipmentType, default: []].append(equipmentItem)`